### ISSUE TITLE
Make fail with missing golangci-lint

### ISSUE THREAD

**Issue by imrehg** [agherzan/git-mirror-me#9]
Pulled the current version (being on 1d7f18c7c6f317fdc4a3c9a7f708378527cd4823), and `make` fails with this error (referring to `golangci-lint`):

```
$ make
go test -race -coverprofile=coverage.out ./...
ok  	github.com/agherzan/git-mirror-me	1.200s	coverage: 85.7% of statements
ok  	github.com/agherzan/git-mirror-me/cmd/git-mirror-me	1.438s	coverage: 80.0% of statements
ok  	github.com/agherzan/git-mirror-me/internal/utils	0.850s	coverage: 82.6% of statements
go tool cover -func=coverage.out
github.com/agherzan/git-mirror-me/cmd/git-mirror-me/args.go:18:	parseArgs		100.0%
github.com/agherzan/git-mirror-me/cmd/git-mirror-me/main.go:16:	run			100.0%
github.com/agherzan/git-mirror-me/cmd/git-mirror-me/main.go:42:	main			0.0%
github.com/agherzan/git-mirror-me/conf.go:40:			GetSSHKey		100.0%
github.com/agherzan/git-mirror-me/conf.go:46:			SetSSHKey		100.0%
github.com/agherzan/git-mirror-me/conf.go:52:			GetKnownHosts		100.0%
github.com/agherzan/git-mirror-me/conf.go:58:			SetKnownHosts		100.0%
github.com/agherzan/git-mirror-me/conf.go:64:			GetKnownHostsPath	100.0%
github.com/agherzan/git-mirror-me/conf.go:70:			SetKnownHostsPath	100.0%
github.com/agherzan/git-mirror-me/conf.go:77:			Pretty			83.3%
github.com/agherzan/git-mirror-me/conf.go:93:			ProcessEnv		100.0%
github.com/agherzan/git-mirror-me/conf.go:115:			Validate		100.0%
github.com/agherzan/git-mirror-me/git.go:32:			filterOutRefs		80.0%
github.com/agherzan/git-mirror-me/git.go:64:			refsToDeleteSpecs	100.0%
github.com/agherzan/git-mirror-me/git.go:75:			extraRefs		92.3%
github.com/agherzan/git-mirror-me/git.go:104:			extraSpecs		75.0%
github.com/agherzan/git-mirror-me/git.go:115:			pruneRemote		72.7%
github.com/agherzan/git-mirror-me/git.go:144:			setupStagingRepo	81.8%
github.com/agherzan/git-mirror-me/git.go:178:			pushWithAuth		76.9%
github.com/agherzan/git-mirror-me/git.go:273:			DoMirror		62.5%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:25:	SortSlice		100.0%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:39:	SlicesAreEqual		100.0%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:57:	NewBareRepo		66.7%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:72:	NewTestRepo		72.7%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:173:	RepoRefsSlice		87.5%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:192:	SpecsToStrings		100.0%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:203:	RefsToStrings		100.0%
github.com/agherzan/git-mirror-me/internal/utils/utils.go:215:	RepoRefsCheckHash	91.7%
github.com/agherzan/git-mirror-me/log.go:27:			NewLogger		100.0%
github.com/agherzan/git-mirror-me/log.go:41:			GetOutput		100.0%
github.com/agherzan/git-mirror-me/log.go:47:			Debug			100.0%
github.com/agherzan/git-mirror-me/log.go:54:			Info			100.0%
github.com/agherzan/git-mirror-me/log.go:59:			Warn			100.0%
github.com/agherzan/git-mirror-me/log.go:64:			Error			100.0%
github.com/agherzan/git-mirror-me/log.go:70:			Fatal			100.0%
github.com/agherzan/git-mirror-me/misc.go:12:			mask			100.0%
total:								(statements)		84.0%
golangci-lint run
make: golangci-lint: No such file or directory
make: *** [lint] Error 1
```

**Comment by imrehg** [comment]
Looking at it, of course this is described in the README, but it's subtle that one needs to install additional tools for it, and looks like it's not really optional, if `make` itself needs it before the `build` step is run?

**Comment by agherzan** [comment]
@imrehg The readme doesn't say it's optional (anymore). I wanted to make sure things are checked before being pushed to the repo as CI requires it. Now about the order, maybe it would be useful to change the order in `make all` or keep `lint` out of it as a separate target. What do you think?

**Comment by imrehg** [comment]
I think it's a tough one, the ultimate solution would be likely something that is included in the project and runs well locally and in CI too -- but that's just moving the goalpost.

Some thoughts I can think of:

* I wouldn't change the order, as it doesn't make to end up with a failure at the end if everything else was okay (and the main purpose is build)
* use `lint` similar to `reuse` (ie. check whether the CLI tool the step uses exists, so making things optional). This is not necessarily great, because it opens the door of silently _not_ running linting, instead of failing, and protecting against that would make the `Makefile` potentially very complicated
* keeping linting as a separate target is probably workable, and CI failures are not inherently bad (fix-errors-as-they-come is quite pragmatic way of thinking about things). I understand how that might offend your sense of order.
* finally, can make the linting tool mandatory for running `make all`, maybe that's the thing that would be the safest and simples (and best way opinionated). Ie. don't change anything (except the README, or warn in the `lint` step if the command doesn't exists to install it). For example for the original report, I should have gone and run `make build` directly and be done with it, to be honest. So it's more user error/inconvenience rather than configuration issue.

TL;DR: After thinking about it, I'd recommend keeping the current method of things for all its benefits, just clarify stuff.

**Comment by agherzan** [comment]
@imrehg What do you think about https://github.com/agherzan/git-mirror-me/pull/12?

**Comment by imrehg** [comment]
Yeah, seems legit, cheers!
